docs/application/dotnet/guides/nui/custom-layout.md

```diff
@@ -0,0 +1,157 @@
+# Create Custom Layout
+
+You can create custom layouts with the fundamental layout of NUI. You can also use the available layouts, such as `LinearLayout` and `GridLayout`. For more information, see the [Common Layouts](./layouts.md#commonLayout) category.
+
+## OnMeasure and OnLayout
+
+Custom layouts can be created to use in applications and also to add to the current **Toolkit**.
+
+The custom layout must be derived from `LayoutGroup` and override the two methods : `OnMeasure` and `OnLayout`.
+
+When creating a custom layout, `OnMeasure` and `OnLayout` methods must be extended and are called during the measure and layout phases respectively. The layout framework does all the heavy work except for measure and layout. Measure and layout are done by Custom Layout.
+
+To create a layout, following are the two phases:
+
+1. `OnMeasure` : Measuring the size requirements of children hence determining the layouts own dimensions.
+
+2. `OnLayout` Laying out and positioning the children within view itself using their measured sizes.
+
+### Measure View and Children
+
+```protected override void OnMeasure( MeasureSpecification widthMeasureSpec, MeasureSpecification heightMeasureSpec )```
+
+The `OnMeasure` is the first function to override the two parameters of width and height of parents. `MeasureSpecifications` imposes the size of the custom layout.
+
+The custom layout must measure its children. To measure, `MeasureChild` API is available along with children, which is a list of the layout’s children. `MeasureChild` can be used to iterate and measure each one. After the children are measured, the custom layout can adjust height and width together as required.
+
+Finally, you should call `SetMeasuredDimensions` API with the size to set the dimensions determined.
```

Comment:
Regarding this comment: https://github.com/Samsung/tizen-docs/pull/991#discussion_r363739155

Are we trying to say:
Finally, you must call the `SetMeasuredDimensions` API with the size to set the required dimensions.

Please verify for technical accuracy and correct me if I am wrong.

Reply:
Done. Okay.